torch/csrc/jit/unpickler.cpp

```diff
@@ -494,6 +494,15 @@ PickleOpCode Unpickler::readInstruction() {
           stack_.back() = IValue();
         });
       } else if (module_name == "torch") {
+        if (class_name == "device") {
+          globals_.emplace_back([this] {
+            auto device_string = stack_.back().toTuple()->elements().at(0);
+            stack_.pop_back();
+            stack_.emplace_back(c10::Device(device_string.toStringRef()));
+          });
+          stack_.emplace_back(int64_t(globals_.size() - 1));
+          return opcode;
```

Comment:
The early returns that have accumulated in this branch of the switch statement should be cleaned up. It is not clear to me why some cases break, others early return, and some push the globals reference number into the stack. Ideally, nothing inside of the switch statement should be returning early.  Perhaps put the global logic in its own function for clarity.